nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -338,15 +343,43 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         session.getProvenanceReporter().receive(flowFile, transitUri);
     }
 
+    @OnPrimaryNodeStateChange
+    public void onPrimaryNodeStateChange(final PrimaryNodeState newState) {
+        if (newState == PrimaryNodeState.PRIMARY_NODE_REVOKED) {
+            stopTweetStreamService(true);
+        }
+    }
+
     @OnStopped
     public void onStopped() {
-        if (tweetStreamService != null) {
-            tweetStreamService.stop();
-        }
-        tweetStreamService = null;
+        stopTweetStreamService(false);
         emptyQueue();
     }
 
+    private synchronized void startTweetStreamService(final ProcessContext context) {
```

Comment:
Is the `synchronized` modifier necessary given that the `AtomicBoolean` is used to track stream status?